src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -285,142 +286,87 @@ public void apply(Project project) {
             });
-            ideIntegration.runTaskOnProjectSync(createLaunchScriptTask);
-
-            tasks.register(InternalModelHelper.nameOfRun(run, "run", ""), RunGameTask.class, task -> {
-                task.setGroup(TASK_GROUP);
-                task.setDescription("Runs the " + run.getName() + " Minecraft run configuration.");
-
-                // Launch with the Java version used in the project
-                var toolchainService = ExtensionUtils.findExtension(project, "javaToolchains", JavaToolchainService.class);
-                task.getJavaLauncher().set(toolchainService.launcherFor(spec -> spec.getLanguageVersion().set(javaExtension.getToolchain().getLanguageVersion())));
-                // Note: this contains both the runtimeClasspath configuration and the sourceset's outputs.
-                // This records a dependency on compiling and processing the resources of the source set.
-                task.getClasspathProvider().from(run.getSourceSet().map(SourceSet::getRuntimeClasspath));
-                task.getGameDirectory().set(run.getGameDirectory());
-
-                task.getEnvironmentProperty().set(run.getEnvironment());
-                task.jvmArgs(RunUtils.getArgFileParameter(prepareRunTask.get().getVmArgsFile().get()).replace("\\", "\\\\"));
-                task.getMainClass().set(RunUtils.DEV_LAUNCH_MAIN_CLASS);
-                task.args(RunUtils.getArgFileParameter(prepareRunTask.get().getProgramArgsFile().get()).replace("\\", "\\\\"));
-                // Of course we need the arg files to be up-to-date ;)
-                task.dependsOn(prepareRunTask);
-                task.dependsOn(run.getTasksBefore());
-
-                task.getJvmArgumentProviders().add(RunUtils.getGradleModFoldersProvider(project, run.getLoadedMods(), null));
-            });
-        });
 
-        setupJarJar(project);
+            configurations.getByName(JavaPlugin.TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME, files -> {
+                files.extendsFrom(configurations.getByName(CONFIGURATION_RUNTIME_DEPENDENCIES));
+                files.extendsFrom(testFixtures);
+            });
 
-        configureTesting = () -> setupTesting(
-                project,
-                modDevBuildDir,
-                userDevConfigOnly,
-                downloadAssets,
-                createArtifacts,
-                neoForgeModDevLibrariesDependency,
-                minecraftClassesArtifact
-        );
+            setupTestTask(
+                    project,
+                    userDevConfigOnly,
+                    tasks.named("test", Test.class),
+                    extension.getUnitTest().getLoadedMods(),
+                    extension.getUnitTest().getTestedMod(),
+                    modDevBuildDir,
+                    modulePath -> {
+                        // NOTE: When running in vanilla mode, this configuration is simply empty
```

Comment:
Uh, copy paste comment?

Reply:
Vanilla isn't relevant when unit testing, this comment is an irrelevant copy paste from the one above :P